controllers/agentcluster_controller.go

```diff
@@ -348,3 +371,37 @@ func (r *AgentClusterReconciler) ensureSecretLabel(ctx context.Context, name, na
 		r.Log.WithError(err).Warn("Failed labeling secret")
 	}
 }
+
+func (r *AgentClusterReconciler) handleDeletion(ctx context.Context, agentCluster *capiproviderv1.AgentCluster) error {
+	if paused := agentCluster.Annotations[clusterv1.PausedAnnotation]; paused == "true" {
+		// unset finalizer, remove owner from owned resources to orphan them and return
+		if err := r.orphanResources(ctx, agentCluster); err != nil {
+			return err
+		}
+	}
+	controllerutil.RemoveFinalizer(agentCluster, agentClusterFinalizer)
+	return nil
+}
+
+func (r *AgentClusterReconciler) orphanResources(ctx context.Context, agentCluster *capiproviderv1.AgentCluster) error {
```

Comment:
This is an interesting approach. I expected us to avoid the owner reference entirely as a part of this change, but I don't see why this wouldn't work as well.

Do you think it's worth removing the owner reference as soon as we see the annotation?

What do you think about this approach @eranco74 ?

Reply:
> Do you think it's worth removing the owner reference as soon as we see the annotation?

Definitely a good consideration.

I think I'd be concerned with removing it prematurely and having to add it back in. For instance, if an AgentCluster was paused and then unpaused without being deleted. We'd have to do another operation to add it back to the cluster deployment.